internal/testutils/basefee/gen_aux_base_fee_test.go

```diff
@@ -0,0 +1,147 @@
+package basefee
+
+import (
+	"context"
+	"fmt"
+	"github.com/taikoxyz/taiko-client/pkg/sender"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/taikoxyz/taiko-client/bindings"
+	"github.com/taikoxyz/taiko-client/internal/testutils"
+	"github.com/taikoxyz/taiko-client/internal/utils"
+)
+
+type BaseFeeSuite struct {
+	testutils.ClientTestSuite
+
+	baseFee *AuxBaseFee
+
+	lastSyncedBlock uint64
+	gasExcess       uint64
+	config          bindings.TaikoL2Config
+}
+
+func (s *BaseFeeSuite) calc1559BaseFee(_l1BlockID, _parentGasUsed uint64) uint64 {
+	var (
+		baseFee *big.Int
+		err     error
+	)
+	if s.gasExcess > 0 {
+		excess := s.gasExcess + _parentGasUsed
+		var numL1Blocks uint64
+		if s.lastSyncedBlock > 0 && _l1BlockID > s.lastSyncedBlock {
+			numL1Blocks = _l1BlockID - s.lastSyncedBlock
+		}
+		if numL1Blocks > 0 {
+			issuance := numL1Blocks * uint64(s.config.GasTargetPerL1Block)
+			if excess > issuance {
+				excess -= issuance
+			} else {
+				excess = 1
+			}
+		}
+		gasExcess := utils.Min(excess, math.MaxUint64)
+		baseFee, err = s.baseFee.BaseFee(nil,
+			new(big.Int).SetUint64(gasExcess),
+			s.config.BasefeeAdjustmentQuotient,
+			s.config.GasTargetPerL1Block,
+		)
+		s.Nil(err)
+	}
+
+	if baseFee == nil || baseFee.Uint64() == 0 {
+		return 1
+	}
+	return baseFee.Uint64()
+}
+
+func (s *BaseFeeSuite) testCalc1559BaseFee(numL1Blocks uint64, startGasUsed uint64, times int) {
+	s.lastSyncedBlock = 1
+	for ; times > 0; times-- {
+		fmt.Println("GasUsed: ", startGasUsed)
+		startGasUsed = s.calc1559BaseFee(s.lastSyncedBlock+numL1Blocks, startGasUsed)
+	}
+}
+
+func (s *BaseFeeSuite) TestCalc1559BaseFee() {
+	var numL1Blocks uint64 = 1
+	times := 10
+	for i := 7740000000; times > 0; i += 400000000 {
+		times--
+		s.testCalc1559BaseFee(numL1Blocks, numL1Blocks*uint64(s.config.GasTargetPerL1Block)+uint64(i), 3)
+	}
+}
+
+func (s *BaseFeeSuite) TestBaseFee() {
```

Comment:
This test case aims to test [baseFee](https://github.com/taikoxyz/taiko-mono/blob/17abc189ca3d2752beb5400c036a650fd5b9c895/packages/protocol/contracts/L2/Lib1559Math.sol#L17) solidity function.